app/lib/frontend/cronjobs.dart

```diff
@@ -1,33 +1,77 @@
+import 'dart:async' show scheduleMicrotask, Completer;
+import 'dart:convert' show utf8;
 import 'package:shelf/shelf.dart' show Request, Response;
 import 'package:logging/logging.dart' show Logger;
 import 'package:googleapis_auth/auth_io.dart' as auth;
 import 'package:googleapis/datastore/v1.dart' as datastore;
 import 'package:ulid/ulid.dart' show Ulid;
+import 'package:gcloud/storage.dart' show Bucket;
+import 'package:stack_trace/stack_trace.dart' show Trace;
+
+import '../shared/configuration.dart' show activeConfiguration;
 
 final _log = new Logger('pub.cronjobs');
 
 class CronJobs {
-  final String backupBucketName;
+  final Bucket backupBucket;
 
-  CronJobs({
-    this.backupBucketName,
-  });
+  CronJobs(this.backupBucket);
 
+  /// Handle a request from the AppEngine cron job scheduler.
+  ///
+  /// Request will succeed if we match it up with a cron-job. This handler
+  /// returns 200 OK, before the cron-job is completed to avoid timeouts from
+  /// the load-balancer. But it keeps the request open by sending spaces until
+  /// the cron-job handler is done, while we won't retry failed jobs, this
+  /// ensures that instances won't be cycled while jobs are running.
+  ///
+  /// Note. it's recommened to configure an alert to fire if a log message from
+  /// the cron job doesn't show at a reasonable interval.
   Future<Response> handler(Request request) async {
-    final job = _resolve(request.url.path);
+    final path = request.requestedUri.path;
+    final job = _resolve(path);
     if (job == null) {
+      _log.shout('### Cron job not found: $path');
       return Response.notFound('no-such-cron-job');
     }
-    _log.finest('### Cron: ${request.url.path}');
-    final ok = await job();
-    if (!ok) {
-      _log.shout('### Cron failed: ${request.url.path}');
-      return Response.internalServerError(
-        body: 'cron failed ${request.url.path}',
-      );
-    }
-    _log.finest('### Cron successful: ${request.url.path}');
-    return Response.ok('cron successful: ${request.url.path}');
+
+    _log.info('### Cron starting: $path');
+    final c = Completer<bool>();
+    scheduleMicrotask(() async {
```

Comment:
why the need for completer + scheduleMicrotask? Wouldn't it be simpler without it?

Reply:
Never mind, I get it.